tests/model_explainability/trustyai_service/utils.py

```diff
@@ -44,27 +49,91 @@ def wait_for_mariadb_operator_deployments(mariadb_operator: MariadbOperator) ->
         deployment.wait_for_replicas()
 
 
-def wait_for_mariadb_pods(client: DynamicClient, mariadb: MariaDB, timeout: int = Timeout.TIMEOUT_5MIN) -> None:
-    def _get_mariadb_pods() -> list[Pod]:
+def get_pods_func(client: DynamicClient, namespace: str, label_selector: Optional[str]) -> Callable[[], list[Pod]]:
+    def _get_pods() -> List[Pod]:
         pods = [
             pod
             for pod in Pod.get(
                 dyn_client=client,
-                namespace=mariadb.namespace,
-                label_selector="app.kubernetes.io/instance=mariadb",
+                namespace=namespace,
+                label_selector=label_selector,
             )
         ]
         return pods
 
-    sampler = TimeoutSampler(wait_timeout=timeout, sleep=1, func=lambda: bool(_get_mariadb_pods()))
+    return _get_pods
+
+
+def wait_for_pods(
+    client: DynamicClient,
+    namespace: str,
+    label_selector: Optional[str] = None,
+    timeout: int = Timeout.TIMEOUT_5MIN,
+    wait_for_condition: str = Pod.Condition.READY,
+    condition_status: str = Pod.Condition.Status.TRUE,
+) -> List[Pod]:
+    get_pods = get_pods_func(client=client, namespace=namespace, label_selector=label_selector)
+    sampler = TimeoutSampler(wait_timeout=timeout, sleep=1, func=lambda: bool(get_pods()))
```

Comment:
Polling every seconds for 5 minutes seems a bit too much. Can we poll every 5 seconds may be?